backend/dataall/base/utils/enum_utils.py

```diff
@@ -0,0 +1,26 @@
+from dataall.base.api import gql
+
+
+def generate_enum_resolver(enum_class):
+    def resolver(context, source):
+        return [{'name': item.name, 'value': item.value} for item in enum_class]
+
+    return resolver
+
+
+EnumItem = gql.ObjectType(
+    name='EnumItem',
+    fields=[
+        gql.Field(name='name', type=gql.String),
+        gql.Field(name='value', type=gql.String),
+    ],
+)
+
+
+def generate_enum_query(enum_class, test_scope):
+    return gql.QueryField(
+        name=enum_class.__name__,
+        type=gql.ArrayType(EnumItem),
```

Comment:
IIRC we have some `int` enums, how are you thinking to handle this? do we have to or we should plan to migrate themt o strings?
should we have have dynamically generated `EnumItem`s something like the following?

```
if isinstance(`enum_class[0].value`, str)
  ...
elif  isinstance(`enum_class[0].value`, int)
  ...
```

Reply:
I think we can just use str(value) since it is passed to frontend.